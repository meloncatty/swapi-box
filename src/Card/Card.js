import React from 'react'
import './Card.css'

const Card = (props) => {
  const peopleCards = ({person_name, homeworld, language, species, population}) => {
    return ( 
      <div>
        <ul>
          <li>
            <h3>{person_name}</h3>
            <button 
              className='favorite' 
              onClick={(e)=> e.preventDefault()}> Favorite
            </button>
          </li>
          <li>Homeworld: {homeworld}</li>
          <li>Population: {population}</li>
          <li>Species: {species}</li>
          <li>Language: {language}</li>
        </ul>
      </div>
    )
  }

  const planetCards = ({planet_name, terrain, population, resident, climate}) => {
    return (
      <div>
        <ul>
          <li>
            <h3>{planet_name}</h3>
            <button 
              className='favorite' 
              onClick={(e)=> e.preventDefault()}> Favorite
            </button>
          </li>
          <li>Terrain: {terrain}</li>
          <li>Population: {population}</li>
          <li>Resident: {resident}</li>
          <li>Climate: {climate}</li>
        </ul>
      </div>
    )
  }

  const vehicleCards = ({vehicle_name, model, passengers, vehicle_class}) => {
    return (
      <div>
        <ul>
          <li>
            <h3>{vehicle_name}</h3>
            <button 
              className='favorite' 
              onClick={(e)=> e.preventDefault()}> Favorite
            </button>
          </li>
          <li>Model: {model}</li>
          <li>Passengers: {passengers}</li>
          <li>Vehicle class: {vehicle_class}</li>
      </ul>
      </div>
    )
  }

  const createCards = (props) => {
    if (props.selectedProps.person_name) {
      return peopleCards(props.selectedProps)
    } else if (props.selectedProps.planet_name) {
      return planetCards(props.selectedProps)
    } else if (props.selectedProps.vehicle_name) {
      return vehicleCards(props.selectedProps)
    }
  }
  
  return (  
    <div>
      {createCards(props)}    
    </div>
  )
}

export default Card
