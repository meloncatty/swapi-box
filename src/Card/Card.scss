.selected-card {
  width: 25%;
  height: 260px;
  margin: 30px;
  letter-spacing: .1em;
}

ul {
  list-style-type: none;
  padding: 0;
}

.add-favorite {
  background: #62ACDD;
  border-radius: 50px;
  border: none;
  width: 15%;
}

.card-header li {
  width: 100%;
  background: #3C6EBC;
  padding: 20px;
  display: flex;
  height: 30px;
  justify-content: space-between;
  align-items: center;
  border-top-left-radius: 25px;
  border-top-right-radius: 25px;
}

.card-header h3 {
  font-size: 2em;
  width: 100%;
}

.card-body {
  padding: 20px;
  background:#B04731;
  color: #E1D6BD;
  width: 100%;
  font-size: 1.5em;
  border-bottom-left-radius: 25px;
  border-bottom-right-radius: 25px;
}

.description {
  color: #000;
}
