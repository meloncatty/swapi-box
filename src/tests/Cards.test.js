import React from 'react'
import {shallow} from 'enzyme'
import Card from '../Card/Card'
import renderer from 'react-test-renderer'

describe('Card', () => {
  it('should match snapshot', () => {
    //setup
    const cardProps = [
      {
        homeworld: 'homeworld',
        species: 'species',
        language: 'language',
        population: 100
      }
    ]

    //execution
    const card = shallow(<Card selectedKeys={cardProps}/>)
    //expectation
    expect(card).toMatchSnapshot()
  });
});
