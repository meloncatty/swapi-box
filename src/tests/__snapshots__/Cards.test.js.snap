// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    selectedKeys={
        Array [
            Object {
              "homeworld": "homeworld",
              "language": "language",
              "population": 100,
              "species": "species",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <ul>
        <li />
</ul>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <li />,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "li",
      },
      "type": "ul",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <ul>
          <li />
</ul>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <li />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "li",
        },
        "type": "ul",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
